zipline/data/history_loader.py

```diff
@@ -141,79 +142,10 @@ def _get_adjustments_in_range(self, asset, dts, field,
-                    end_loc = dts.searchsorted(dt)
-                    adj_loc = end_loc
-                    if is_perspective_after:
-                        # Set adjustment pop location so that it applies
-                        # to last value if adjustment occurs immediately after
-                        # the last slot.
-                        adj_loc -= 1
-                    mult = Float64Multiply(0,
-                                           end_loc - 1,
-                                           0,
-                                           0,
-                                           m[1])
-                    try:
-                        adjs[adj_loc].append(mult)
-                    except KeyError:
-                        adjs[adj_loc] = [mult]
-            divs = self._adjustments_reader.get_adjustments_for_sid(
-                'dividends', sid)
-            for d in divs:
-                dt = d[0]
-                if start < dt <= end:
-                    end_loc = dts.searchsorted(dt)
-                    adj_loc = end_loc
-                    if is_perspective_after:
-                        # Set adjustment pop location so that it applies
-                        # to last value if adjustment occurs immediately after
-                        # the last slot.
-                        adj_loc -= 1
-                    mult = Float64Multiply(0,
-                                           end_loc - 1,
-                                           0,
-                                           0,
-                                           d[1])
-                    try:
-                        adjs[adj_loc].append(mult)
-                    except KeyError:
-                        adjs[adj_loc] = [mult]
-        splits = self._adjustments_reader.get_adjustments_for_sid(
-            'splits', sid)
-        for s in splits:
-            dt = s[0]
-            if start < dt <= end:
-                if field == 'volume':
-                    ratio = 1.0 / s[1]
-                else:
-                    ratio = s[1]
-                end_loc = dts.searchsorted(dt)
-                adj_loc = end_loc
-                if is_perspective_after:
-                    # Set adjustment pop location so that it applies
-                    # to last value if adjustment occurs immediately after
-                    # the last slot.
-                    adj_loc -= 1
-                mult = Float64Multiply(0,
-                                       end_loc - 1,
-                                       0,
-                                       0,
-                                       ratio)
-                try:
-                    adjs[adj_loc].append(mult)
-                except KeyError:
-                    adjs[adj_loc] = [mult]
-        return adjs
+        end_offset = -1
+        date_offset = -1 if is_perspective_after else 0
+        return self._adjustments_reader.load_adjustments(
```

Comment:
Yes, I anticipate making that change on the path to supporting continuous futures.